packages/atomic/src/utils/props-utils.ts

```diff
@@ -15,20 +17,28 @@ export function MapProp(opts?: MapPropOptions) {
     }
 
     component.componentWillLoad = function () {
-      const prefix = (opts && opts.attributePrefix) || variableName;
-      const variable = this[variableName];
+      const prefix = opts?.attributePrefix ?? variableName;
       const attributes = getElement(this).attributes;
-      mapAttributesToProp(prefix, variable, Array.from(attributes));
+      const map = attributesToStringMap(prefix, Array.from(attributes));
+      this[variableName] = opts?.splitValues
+        ? stringMapToStringArrayMap(map)
+        : map;
```

Comment:
Rather than exporting 2 methods for tests purposes, stringMapToStringArrayMap & attributesToStringMap, this piece of logic could be kept in the exported mapAttributesToProp which would be able to handle slipping values or not. stringMapToStringArrayMap & attributesToStringMap would still be as is but not exported